src/simulation_engine.cpp

```diff
@@ -1,21 +1,92 @@
 #include <iostream>
 #include <vector>
 #include <memory>
+#include <random>
+#include <algorithm>
+#include <iterator>
 
 #include <simulator/simulation_engine.hpp>
 #include <simulator/simulation_map.hpp>
+#include <simulator/bug.hpp>
 
 namespace simulation {
 
-SimulationEngine::SimulationEngine() {}
+const uint16_t ANTS = 10;
+const uint16_t DOODLEBUGS = 3;
 
-void SimulationEngine::simulate_round() {}
+SimulationEngine::SimulationEngine(): map_(std::make_shared<SimulationMap>()){}
 
-void SimulationEngine::populate_map_with_species() {}
+void SimulationEngine::start_simulation(uint16_t n_rounds) {
+  for(;n_rounds && simulate_round();n_rounds--);
```

Comment:
Submit code [here](http://www.ioccc.org/) :laughing:

Reply:
I like that line too much, will add a comment for clarity but I want to keep it hahah